src/partition/src/splitter.rs

```diff
@@ -42,12 +43,35 @@ impl WriteSplitter {
         }
     }
 
-    pub fn split_insert(&self, insert: InsertRequest) -> Result<InsertRequestSplit> {
+    pub fn split_insert(
+        &self,
+        insert: InsertRequest,
+        schema: &Schema,
+    ) -> Result<InsertRequestSplit> {
         check_req(&insert)?;
-
-        let column_names = self.partition_rule.partition_columns();
-        let values = &insert.columns_values;
-        let partition_columns = find_partitioning_values(values, &column_names)?;
+        let mut insert = insert;
+        let partition_columns = self.partition_rule.partition_columns();
+        for column_schema in schema.column_schemas() {
+            if partition_columns.contains(&column_schema.name)
+                && (!insert.columns_values.contains_key(&column_schema.name))
+            {
+                let key = insert.columns_values.keys().next().unwrap();
+                let row_nums = insert.columns_values.get(key).unwrap().len();
+                let default_values = column_schema
+                    .create_default_vector(row_nums)
+                    .context(CreateDefaultToReadSnafu {
+                        column: &column_schema.name,
+                    })?
+                    .context(MissingDefaultValueSnafu {
+                        column: &column_schema.name,
+                    })?;
+                insert
+                    .columns_values
+                    .insert(column_schema.name.clone(), default_values);
+            }
```

Comment:
I get the following error:

```
cannot borrow `insert.columns_values` as mutable because it is also borrowed as immutable
mutable borrow occurs hererustc ~
//...
splitter.rs(57, 21): immutable borrow occurs here
splitter.rs(57, 73): first borrow occurs due to use of `insert.columns_values` in closure
splitter.rs(59, 30): immutable borrow later used here
```

So, how about this instead?

```rust
        let partition_column_schemas = schema
            .column_schemas()
            .iter()
            .filter(|schema| partition_columns.contains(&schema.name));
        for partition_column_schema in partition_column_schemas {
            if !insert.columns_values.contains_key(&partition_column_schema.name){
                let default_values = partition_column_schema
                    .create_default_vector(row_nums)
                    .context(CreateDefaultToReadSnafu {
                        column: &partition_column_schema.name,
                    })?
                    .context(MissingDefaultValueSnafu {
                        column: &partition_column_schema.name,
                    })?;
                insert
                    .columns_values
                    .insert(partition_column_schema.name.clone(), default_values);
            }
        }
```

Reply:
> I get the following error:

That was my mistake. Try this. It's worked for me
```rust
            .filter(|schema| partition_columns.contains(&schema.name) && !insert.columns_values.contains_key(&schema.name))
            .collect::<Vec<_>>();
```